fasthtml/js.py

```diff
@@ -87,3 +88,47 @@ def SortableJS(
 proc_htmx('%s', Sortable.create);
 """ % sel
     return Script(src, type='module')
+
+# %% ../nbs/api/03_js.ipynb
+def MermaidJS(
+        sel='.language-mermaid',  # CSS selector for mermaid elements
+        theme='base',  # Mermaid theme to use
+        delay=500  # Delay in milliseconds before rendering
+    ):
+    "Implements browser-based Mermaid diagram rendering."
+    src = """
+import mermaid from 'https://cdn.jsdelivr.net/npm/mermaid@11/dist/mermaid.esm.min.mjs';
+
+mermaid.initialize({
+    startOnLoad: false,
+    theme: '%s',
+    securityLevel: 'loose',
+    flowchart: { useMaxWidth: false, useMaxHeight: false }
+});
+
+function renderMermaidDiagrams() {
+    const diagrams = document.querySelectorAll('%s');
+    diagrams.forEach((element, index) => {
+        try {
+            const graphDefinition = element.textContent;
+            const graphId = `mermaid-diagram-${index}`;
+            mermaid.render(graphId, graphDefinition)
+                .then(({svg, bindFunctions}) => {
+                    element.innerHTML = svg;
+                    bindFunctions?.(element);
+                })
+                .catch(error => {
+                    console.error(`Error rendering Mermaid diagram ${index}:`, error);
+                    element.innerHTML = `<p>Error rendering diagram: ${error.message}</p>`;
+                });
+        } catch (error) {
+            console.error(`Error processing Mermaid diagram ${index}:`, error);
+        }
+    });
+}
+
+proc_htmx('%s', () => {
+    setTimeout(renderMermaidDiagrams, %d);
+});
+""" % (theme, sel, sel, delay)
```

Comment:
While the JS in this file should be in its own file, I will recommend that 99% of the time f-strings are better than this kind of string variable injection. The reason is clarity, which makes debugging easier. The 1% edge case would be to increase performance, however in web dev strings are rarely the bottleneck.

Reply:
the benefit of py2 style formatting here is that you don't have to deal with doubling the curly brackets.  it's a minor thing but can be more readable